### PR TITLE
ticdc: add doc about config

### PR DESCRIPTION
**Pull request by overvenus**
### What is changed, added or deleted? (Required)

Add doc about TiCDC config.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: N/A
- Other reference link(s): N/A

### CODE REVIEW

File: ticdc/manage-ticdc.md [hunk 1]
```diff
@@ -351,6 +363,54 @@ curl -X POST http://127.0.0.1:8301/capture/owner/resign
...
+# 需要过滤掉的复制 ID
+filter-replica-ids = [2,3]
+# 是否同步 DDL
+sync-ddl = true
+```
+
+### 配置文件兼容注意事项
+
+* 4.0.0 移除了 `ignore-txn-commit-ts`，添加了 `ignore-txn-start-ts`，使用 start ts 过滤事务。
+* 4.0.0 移除了 `db-dbs`/`db-tables`/`ignore-dbs`/`ignore-tables`，添加了 `rules`，使用新版的数据库数据表过滤规则，[详细语法参考](https://github.com/pingcap/tidb-tools/blob/master/pkg/table-filter/README.md)。
```
Comment (amyangfei) [inline ticdc/manage-ticdc.md:412]:
```suggestion
* 4.0.2 移除了 `db-dbs`/`db-tables`/`ignore-dbs`/`ignore-tables`，添加了 `rules`，使用新版的数据库数据表过滤规则，[详细语法参考](https://github.com/pingcap/tidb-tools/blob/master/pkg/table-filter/README.md)。
```

**Comment by TomShawn** [comment]
@overvenus Please resolve the conflict.

File: ticdc/manage-ticdc.md [hunk 2]
```diff
@@ -150,6 +150,18 @@ create changefeed ID: 28c43ffc-2316-4f4f-a70b-d1a7c59ba79f info {"sink-uri":"mys
     | `max-message-bytes`  | 每次向 Kafka broker 发送消息的最大数据量（可选，默认值 `64MB`） |
     | `replication-factor` | kafka 消息保存副本数（可选，默认值 `1`）                       |
```
Comment (leoppro) [inline ticdc/manage-ticdc.md:151]:
```suggestion
    | `replication-factor` | kafka 消息保存副本数（可选，默认值 `1`）                       |
    | `protocol` | 输出到 kafka 消息协议，可选值 default, canal（可选，默认值 `default`）|
```

File: ticdc/manage-ticdc.md [hunk 3]
```diff
@@ -149,6 +149,19 @@ create changefeed ID: 28c43ffc-2316-4f4f-a70b-d1a7c59ba79f info {"sink-uri":"mys
     | `partition-num`      | 下游 Kafka partition 数量（可选，不能大于实际 partition 数量。如果不填会自动获取 partition 数量。） |
     | `max-message-bytes`  | 每次向 Kafka broker 发送消息的最大数据量（可选，默认值 `64MB`） |
     | `replication-factor` | kafka 消息保存副本数（可选，默认值 `1`）                       |
+    | `protocol` | 输出到 kafka 消息协议，可选值 default, canal（可选，默认值 `default`）    |
```
Comment (TomShawn) [inline ticdc/manage-ticdc.md:152]:
```suggestion
    | `protocol` | 输出到 kafka 消息协议，可选值有 `default`, `canal`（默认值为 `default`）    |
```

File: ticdc/manage-ticdc.md [hunk 4]
```diff
@@ -149,6 +149,19 @@ create changefeed ID: 28c43ffc-2316-4f4f-a70b-d1a7c59ba79f info {"sink-uri":"mys
...
+
+使用方法如下：
+
+{{< copyable "shell-regular" >}}
+
+```shell
+cdc cli changefeed create --pd=http://10.0.10.25:2379 --sink-uri="mysql://root:123456@127.0.0.1:3306/" --config changefeed.toml
+```
+
+其中 “changefeed.toml” 为同步任务的配置文件。
```
Comment (TomShawn) [inline ticdc/manage-ticdc.md:164]:
```suggestion
其中 `changefeed.toml` 为同步任务的配置文件。
```

File: ticdc/manage-ticdc.md [hunk 5]
```diff
@@ -358,6 +371,54 @@ curl -X POST http://127.0.0.1:8301/capture/owner/resign
 election: not leader
 ```
 
+## 同步任务配置文件描述
+
+以下内容详细介绍了同步任务的配置。
+
+```toml
+# 指定配置文件中涉及的库名、表名是否为大小写敏感的
```
Comment (TomShawn) [inline ticdc/manage-ticdc.md:379]:
```suggestion
# 指定配置文件中涉及的库名、表名是否为大小写敏感
```

File: ticdc/manage-ticdc.md [hunk 6]
```diff
@@ -358,6 +371,54 @@ curl -X POST http://127.0.0.1:8301/capture/owner/resign
...
+# 该配置会同时影响 filter 和 sink 相关配置，默认为 true
+case-sensitive = true
+
+[filter]
+# 忽略指定 StartTs 的事务
```
Comment (TomShawn) [inline ticdc/manage-ticdc.md:384]:
```suggestion
# 忽略指定 star_ts 的事务
```

File: ticdc/manage-ticdc.md [hunk 7]
```diff
@@ -358,6 +371,54 @@ curl -X POST http://127.0.0.1:8301/capture/owner/resign
...
+ignore-txn-start-ts = [1, 2]
+
+# 过滤器规则
+# 过滤规则语法： https://github.com/pingcap/tidb-tools/tree/master/pkg/table-filter#syntax
```
Comment (TomShawn) [inline ticdc/manage-ticdc.md:388]: @YiniXu9506 新版官网是否支持在注释中添加网址呢？
```suggestion
# 过滤规则语法：https://github.com/pingcap/tidb-tools/tree/master/pkg/table-filter#syntax
```

File: ticdc/manage-ticdc.md [hunk 8]
```diff
@@ -358,6 +371,54 @@ curl -X POST http://127.0.0.1:8301/capture/owner/resign
...
+rules = ['*.*', '!test.*']
+
+[mounter]
+# mounter 线程数，用于解码 TiKV 输出的数据
+worker-num = 16
+
+[sink]
+# 对于 MQ 类的 Sink，可以通过 dispatchers 配置 event 分发器
+# 分发器支持 default, ts, rowid, table 四种
```
Comment (TomShawn) [inline ticdc/manage-ticdc.md:397]: 分发器支持这四种什么呢？值还是对象还是...？
```suggestion
# 分发器支持 default、ts、rowid、table 四种
```

File: ticdc/manage-ticdc.md [hunk 9]
```diff
@@ -358,6 +371,54 @@ curl -X POST http://127.0.0.1:8301/capture/owner/resign
...
+dispatchers = [
+    {matcher = ['test1.*', 'test2.*'], dispatcher = "ts"},
+    {matcher = ['test3.*', 'test4.*'], dispatcher = "rowid"},
+]
+# 对于 MQ 类的 Sink，可以指定消息的协议格式
+# 协议目前支持 default, canal 两种，default 为 ticdc-open-protocol
```
Comment (TomShawn) [inline ticdc/manage-ticdc.md:403]:
```suggestion
# 目前支持 default 和 canal 两种协议。default 为 TiCDC Open Protocol
```

File: ticdc/manage-ticdc.md [hunk 10]
```diff
@@ -358,6 +371,54 @@ curl -X POST http://127.0.0.1:8301/capture/owner/resign
...
+protocol = "default"
+
+[cyclic-replication]
+# 是否开启环形复制
+enable = false
+# 当前 CDC 的复制 ID
```
Comment (TomShawn) [inline ticdc/manage-ticdc.md:409]: `复制`还是`同步`？
```suggestion
# 当前 TiCDC 的复制 ID
```

File: ticdc/manage-ticdc.md [hunk 11]
```diff
@@ -358,6 +371,54 @@ curl -X POST http://127.0.0.1:8301/capture/owner/resign
...
+dispatchers = [
+    {matcher = ['test1.*', 'test2.*'], dispatcher = "ts"},
+    {matcher = ['test3.*', 'test4.*'], dispatcher = "rowid"},
+]
+# 对于 MQ 类的 Sink，可以指定消息的协议格式
+# 协议目前支持 default, canal 两种，default 为 ticdc-open-protocol
+protocol = "default"
+
+[cyclic-replication]
+# 是否开启环形复制
```
Comment (TomShawn) [inline ticdc/manage-ticdc.md:407]: 环形`同步`还是`复制`呢？

File: ticdc/manage-ticdc.md [hunk 12]
```diff
@@ -358,6 +371,54 @@ curl -X POST http://127.0.0.1:8301/capture/owner/resign
...
+enable = false
+# 当前 CDC 的复制 ID
+replica-id = 1
+# 需要过滤掉的复制 ID
+filter-replica-ids = [2,3]
+# 是否同步 DDL
+sync-ddl = true
+```
+
+### 配置文件兼容注意事项
```
Comment (TomShawn) [inline ticdc/manage-ticdc.md:417]:
```suggestion
### 配置文件兼容性的注意事项
```

File: ticdc/manage-ticdc.md [hunk 13]
```diff
@@ -358,6 +371,54 @@ curl -X POST http://127.0.0.1:8301/capture/owner/resign
...
+
+* 4.0.0 移除了 `ignore-txn-commit-ts`，添加了 `ignore-txn-start-ts`，使用 start ts 过滤事务。
```
Comment (TomShawn) [inline ticdc/manage-ticdc.md:419]:
```suggestion
* TiDB v4.0.0 中移除了 `ignore-txn-commit-ts`，添加了 `ignore-txn-start-ts`，使用 start_ts 过滤事务。
```

File: ticdc/manage-ticdc.md [hunk 14]
```diff
@@ -358,6 +371,54 @@ curl -X POST http://127.0.0.1:8301/capture/owner/resign
...
+* 4.0.2 移除了 `db-dbs`/`db-tables`/`ignore-dbs`/`ignore-tables`，添加了 `rules`，使用新版的数据库数据表过滤规则，[详细语法参考](https://github.com/pingcap/tidb-tools/blob/master/pkg/table-filter/README.md)。
```
Comment (TomShawn) [inline ticdc/manage-ticdc.md:420]:
```suggestion
* TiDB v4.0.2 中移除了 `db-dbs`/`db-tables`/`ignore-dbs`/`ignore-tables`，添加了 `rules`，使用新版的数据库和数据表过滤规则，详细语法参考 [Table Filter](https://github.com/pingcap/tidb-tools/blob/master/pkg/table-filter/README.md)。
```

File: ticdc/manage-ticdc.md [hunk 15]
```diff
@@ -358,6 +371,54 @@ curl -X POST http://127.0.0.1:8301/capture/owner/resign
...
+# 对于 MQ 类的 Sink，可以指定消息的协议格式
+# 协议目前支持 default, canal 两种，default 为 ticdc-open-protocol
+protocol = "default"
+
+[cyclic-replication]
+# 是否开启环形复制
+enable = false
+# 当前 CDC 的复制 ID
+replica-id = 1
+# 需要过滤掉的复制 ID
```
Comment (TomShawn) [inline ticdc/manage-ticdc.md:411]: ditto

File: ticdc/manage-ticdc.md (hunk 6 above)
Comment (overvenus) [reply ticdc/manage-ticdc.md:384]: "start ts" is a term in tidb.

File: ticdc/manage-ticdc.md (hunk 11 above)
Comment (overvenus) [reply ticdc/manage-ticdc.md:407]: “同步”

Comment (overvenus) [inline ticdc/manage-ticdc.md:407]:
```suggestion
# 是否开启环形同步
```

File: ticdc/manage-ticdc.md (hunk 15 above)
Comment (overvenus) [inline ticdc/manage-ticdc.md:411]:
```suggestion
# 需要过滤掉的同步 ID
```

File: ticdc/manage-ticdc.md (hunk 6 above)
Comment (overvenus) [inline ticdc/manage-ticdc.md:384]:
```suggestion
# 忽略指定 start_ts 的事务
```

File: ticdc/manage-ticdc.md (hunk 14 above)
Comment (overvenus) [inline ticdc/manage-ticdc.md:420]:
```suggestion
* TiCDC v4.0.2 中移除了 `db-dbs`/`db-tables`/`ignore-dbs`/`ignore-tables`，添加了 `rules`，使用新版的数据库和数据表过滤规则，详细语法参考 [Table Filter](https://github.com/pingcap/tidb-tools/blob/master/pkg/table-filter/README.md)。
```

File: ticdc/manage-ticdc.md (hunk 13 above)
Comment (overvenus) [inline ticdc/manage-ticdc.md:419]:
```suggestion
* TiCDC v4.0.0 中移除了 `ignore-txn-commit-ts`，添加了 `ignore-txn-start-ts`，使用 start_ts 过滤事务。
```

**Comment by overvenus** [comment]
PTAL @TomShawn , thanks!

File: ticdc/manage-ticdc.md [hunk 16]
```diff
@@ -358,6 +371,54 @@ curl -X POST http://127.0.0.1:8301/capture/owner/resign
...
+# 过滤规则语法：https://github.com/pingcap/tidb-tools/tree/master/pkg/table-filter#syntax
+rules = ['*.*', '!test.*']
+
+[mounter]
+# mounter 线程数，用于解码 TiKV 输出的数据
+worker-num = 16
+
+[sink]
+# 对于 MQ 类的 Sink，可以通过 dispatchers 配置 event 分发器
+# 分发器支持 default、ts、rowid、table 四种
```
Comment (TomShawn) [inline ticdc/manage-ticdc.md:397]:
```suggestion
# 支持 default、ts、rowid、table 四种分发器
```

File: ticdc/manage-ticdc.md [hunk 17]
```diff
@@ -149,6 +149,19 @@ create changefeed ID: 28c43ffc-2316-4f4f-a70b-d1a7c59ba79f info {"sink-uri":"mys
     | `partition-num`      | 下游 Kafka partition 数量（可选，不能大于实际 partition 数量。如果不填会自动获取 partition 数量。） |
     | `max-message-bytes`  | 每次向 Kafka broker 发送消息的最大数据量（可选，默认值 `64MB`） |
     | `replication-factor` | kafka 消息保存副本数（可选，默认值 `1`）                       |
+    | `protocol` | 输出到 kafka 消息协议，可选值有 `default`, `canal`（默认值为 `default`）    |
+
+如需设置更多同步任务的配置，比如指定同步单个数据表，请参阅[同步任务配置文件描述](#同步任务配置文件描述)。
+
+使用方法如下：
```
Comment (TomShawn) [inline ticdc/manage-ticdc.md:156]: 什么的使用方法呢？建议具体一些。Rest LGTM

Comment (overvenus) [inline ticdc/manage-ticdc.md:156]:
```suggestion
使用配置文件创建同步任务的方法如下：
```

Comment (overvenus) [reply ticdc/manage-ticdc.md]: 已更新，麻烦尽快 review 下

File: ticdc/troubleshoot-ticdc.md [hunk 18]
```diff
@@ -109,3 +109,7 @@ cdc cli changefeed create --sink-uri="mysql://root@127.0.0.1:3306/" --tz=Asia/Sh
 > - 最优先使用 `--tz` 传入的时区
 > - 没有 `--tz` 参数，会尝试读取 `TZ` 环境变量设置的时区
 > - 如果还没有 `TZ` 环境变量，会从 TiCDC server 运行机器的默认时区
+
+## 升级 TiCDC 后配置文件不兼容
+
+请参阅[配置文件兼容注意事项](/ticdc/manage-ticdc.md#配置文件兼容注意事项)。
```
Comment (TomShawn) [inline ticdc/troubleshoot-ticdc.md:115]:
```suggestion
请参阅[配置文件兼容注意事项](/ticdc/manage-ticdc.md##配置文件兼容性的注意事项)。
```

**Pull request merged**